### ISSUE TITLE
can not connect to opensearch as of application .env openhost connection env var auto created but not apply to .env file

### ISSUE THREAD

**Issue by jack-yu-matrix** [aws-samples/generative-bi-using-rag#326]
for example:
OPENSEARCH_SECRETS_URL_HOST=opensearch-host-url-e58525
OPENSEARCH_SECRETS_USERNAME_PASSWORD=opensearch-master-user-e58525

 const OSMasterUserSecretNamePrefix = 'opensearch-master-user'; // Add the secret name here
    const guid = crypto.randomBytes(3).toString('hex');
    this.OSMasterUserSecretName = `${OSMasterUserSecretNamePrefix}-${guid}`;

OSMasterUserSecretName are different , so need to manual rebuild docker file , can not use cdk to apply changes

**Comment by jack-yu-matrix** [comment]
the OPENSEARCH_SECRETS_URL_HOST and OPENSEARCH_SECRETS_USERNAME_PASSWORD will be generated dynamically during each time of cdk deploy
1. generate open search stack
2. copy dynamically generated OPENSEARCH_SECRETS_URL_HOST OPENSEARCH_SECRETS_USERNAME_PASSWORD to .env file
3. deploy ecs stack, at this time point the opensearch stack will be update and generated a new host url and secrets username, which resulting in a loop th update .env file under application folder

**Comment by jack-yu-matrix** [comment]
my workaround solution like this :

 const OSMasterUserSecretNamePrefix = 'opensearch-master-user'; // Add the secret name here
    //const guid = crypto.randomBytes(3).toString('hex');
    const guid = 'capgemini-chatbi'

**Comment by jack-yu-matrix** [comment]
streamlit app still got below error:
File "/usr/local/lib/python3.10/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 584, in _run_script
    exec(code, module.__dict__)
File "/app/pages/mainpage.py", line 36, in <module>
    VectorStore.add_entity_sample(current_profile, entity, comment)
File "/app/nlq/business/vector_store.py", line 95, in add_entity_sample
    has_same_sample = cls.search_same_query(profile_name, 5, opensearch_info['ner_index'], embedding)
File "/app/nlq/business/vector_store.py", line 203, in search_same_query
    search_res = cls.search_sample_with_embedding(profile_name, top_k, index_name, embedding)
File "/app/nlq/business/vector_store.py", line 198, in search_sample_with_embedding
    sample_list = cls.opensearch_dao.search_sample_with_embedding(profile_name, top_k, index_name, query_embedding)
File "/app/nlq/data_access/opensearch.py", line 241, in search_sample_with_embedding
    response = self.opensearch_client.search(
File "/usr/local/lib/python3.10/site-packages/opensearchpy/client/utils.py", line 181, in _wrapped
    return func(*args, params=params, headers=headers, **kwargs)
File "/usr/local/lib/python3.10/site-packages/opensearchpy/client/__init__.py", line 1742, in search
    return self.transport.perform_request(
File "/usr/local/lib/python3.10/site-packages/opensearchpy/transport.py", line 448, in perform_request
    raise e
File "/usr/local/lib/python3.10/site-packages/opensearchpy/transport.py", line 409, in perform_request
    status, headers_response, data = connection.perform_request(
File "/usr/local/lib/python3.10/site-packages/opensearchpy/connection/http_urllib3.py", line 290, in perform_request
    self._raise_error(
File "/usr/local/lib/python3.10/site-packages/opensearchpy/connection/base.py", line 316, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(

**Comment by supinyu** [comment]
Hello，Thank you for raising the issue.

This issue will be fixed in version 1.8 by changing the guid to VPC ID. Thank you.

**Comment by supinyu** [comment]
In the branch v.1.7.0 and v.1.8.0， in the file:source/resources/lib/aos/aos-stack.ts

The code has been modified to the following format

```
    const vpcIdSuffix = props.vpc.vpcId
    console.log(`VPC ID Suffix: ${vpcIdSuffix}`);
    this.OSMasterUserSecretName = `${OSMasterUserSecretNamePrefix}-${vpcIdSuffix}`;
```

Thank you.
